### ISSUE TITLE
Segmentation Fault ruby 2.7.1

### ISSUE THREAD

**Issue by RailsCod3rFuture** [tangledpath/ruby-fann#22]
I don't know if this gem is compatible with ruby 2.7.0+. But, i am receiving a massive segmentation fault when I attempt to run the rb file. Can someone please test this gem for ruby 2.7.1. I want to make sure that I'm not doing something wrong.

**Comment by yzgan** [comment]
I'm using ruby 3.0.0p0 (2020-12-25 revision 95aff21468) [x86_64-linux] on WSL and having the same errors when trying to train data.
```
[BUG] Segmentation fault at 0x0000000000000003
```

**Comment by lukeovalle** [comment]
Same problem in `ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-linux-gnu]` on OpenSuse

`red_neuronal.rb:25: [BUG] Segmentation fault at 0x0000000000000003
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x86_64-linux-gnu]

-- Control frame information -----------------------------------------------
c:0004 p:---- s:0026 e:000025 CFUNC  :initialize
c:0003 p:---- s:0023 e:000022 CFUNC  :new
c:0002 p:0133 s:0018 E:000740 EVAL   red_neuronal.rb:25 [FINISH]
c:0001 p:0000 s:0003 E:000890 (none) [FINISH]

-- Ruby level backtrace information ----------------------------------------
red_neuronal.rb:25:in `<main>'
red_neuronal.rb:25:in `new'
red_neuronal.rb:25:in `initialize'

-- Machine register context ------------------------------------------------
 RIP: 0x00007f765cd77910 RBP: 0x0000000000000002 RSP: 0x00007ffc1749d200
 RAX: 0x0000000000000003 RBX: 0x000055b6e0c44b08 RCX: 0x0000000000000006
 RDX: 0x0000000000000000 RDI: 0x000055b6e0c3e470 RSI: 0x000055b6e0c44b08
  R8: 0x000055b6e0c80780  R9: 0xfeb93d2de6772801 R10: 0xfffffffffffff126
 R11: 0x00007f766088d8d0 R12: 0x000055b6e0c44ae0 R13: 0x00007f7660336f68
 R14: 0x000055b6e0c80780 R15: 0x000055b6e0b0ff80 EFL: 0x0000000000010246

-- C level backtrace information -------------------------------------------
[1]    5233 segmentation fault (core dumped)  ruby red_neuronal.rb`

**Comment by rohanaceres** [comment]
Same problem:

```
C:/Ruby27-x64/lib/ruby/gems/2.7.0/gems/gapic-common-0.3.4/lib/gapic/grpc/service_stub.rb:73: [BUG] Segmentation fault
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x64-mingw32]
```

**Comment by shamim-42** [comment]
Hi there,
I am having similar problem. Did anyone find any solution ?

**Comment by Sandtod** [comment]
The same problem is with:
ruby 2.7.2p137 (2020-10-01 revision 5445e04352) [x64-mingw32]
ruby 3.1.3p185 (2022-11-24 revision 1a6b16756e) [x86_64-linux]

**Comment by jb-ht** [comment]
same problem with 
```
ruby -v
ruby 3.2.2 (2023-03-30 revision e51014f9c0) [arm64-darwin22]
```

**Comment by dpblnt** [comment]
```
$ irb
irb(main):001:0> require 'ruby-fann'
=> true
irb(main):002:0> train = RubyFann::TrainData.new(:inputs => [[0.3, 0.4, 0.5], [0.1, 0.2, 0.3]], :desired_outputs => [0.7, 0.8])
(irb):2: [BUG] Segmentation fault at 0x0000000000000000
ruby 3.1.4p223 (2023-03-30 revision 957bb7cb81) [x86_64-linux]

-- Control frame information -----------------------------------------------
c:0022 p:---- s:0113 e:000112 CFUNC  :initialize
c:0021 p:---- s:0110 e:000109 CFUNC  :new
c:0020 p:0023 s:0105 E:000cc8 EVAL   (irb):2 [FINISH]
```

**Comment by dpblnt** [comment]
ruby 2.6.6p146 and ruby-fann (1.3.2)
```
$ irb
irb(main):001:0> require 'ruby-fann'
=> true
irb(main):002:0> train = RubyFann::TrainData.new(:inputs => [[0.3, 0.4, 0.5], [0.1, 0.2, 0.3]], :desired_outputs => [0.7, 0.8])
FANN Error 11: Unable to allocate memory.
=> #<RubyFann::TrainData:0x0085f458>
irb(main):003:0> 

```

**Comment by git-steven** [comment]
Fixed in v2.0
